eng/emitters/pipelines/templates/archetype-emitter.yml

```diff
@@ -68,58 +68,46 @@ extends:
       #   packages/: Contains the packages to publish
       - stage: Build
         variables:
+          -template: /eng/common/pipelines/templates/variables/node-matrix.yml@self
```

Comment:
Instead of trying to declare the matrix as variable you could declare it as a parameter in this template with the default value of what you want.